cmd/testworkflow-toolkit/env/client.go

```diff
@@ -88,25 +195,35 @@ func Testkube() client.Client {
 }
 
 var (
-	cloudMu       sync.Mutex
-	cloudExecutor cloudexecutor.Executor
-	cloudClient   cloud.TestKubeCloudAPIClient
+	cloudMu     sync.Mutex
+	cloudClient cloud.TestKubeCloudAPIClient
+	cloudConn   *grpc.ClientConn
 )
 
-func Cloud(ctx context.Context) (cloudexecutor.Executor, cloud.TestKubeCloudAPIClient) {
+func CloudInternal() cloud.TestKubeCloudAPIClient {
 	cloudMu.Lock()
 	defer cloudMu.Unlock()
 
-	if cloudExecutor == nil {
+	var err error
+	if cloudClient == nil {
 		cfg := config2.Config().Worker.Connection
 		logger := log.NewSilent()
-		grpcConn, err := agentclient.NewGRPCConnection(ctx, cfg.TlsInsecure, cfg.SkipVerify, cfg.Url, "", "", "", logger)
+		cloudConn, err = agentclient.NewGRPCConnection(context.Background(), cfg.TlsInsecure, cfg.SkipVerify, cfg.Url, "", "", "", logger)
 		if err != nil {
 			ui.Fail(fmt.Errorf("failed to connect with Cloud: %w", err))
 		}
-		cloudClient = cloud.NewTestKubeCloudAPIClient(grpcConn)
-		cloudExecutor = cloudexecutor.NewCloudGRPCExecutor(cloudClient, grpcConn, cfg.ApiKey)
+		cloudClient = cloud.NewTestKubeCloudAPIClient(cloudConn)
 	}
+	return cloudClient
+}
 
-	return cloudExecutor, cloudClient
+func Cloud() controlplaneclient.ExecutionSelfClient {
+	cfg := config2.Config()
+	grpcClient := CloudInternal()
+	loadProContext() // FIXME: do it lazily
+	return controlplaneclient.New(grpcClient, internalProContext, cfg.Worker.Connection.AgentID, cfg.Worker.Connection.ApiKey, controlplaneclient.ClientOptions{
+		StorageSkipVerify:  true, // FIXME?
```

Comment:
should it be taken from config?

Reply:
In general - I think so. It's in context in Toolkit though, and before these changes it was hardcoded in every individual place to `true`, so better to not touch it for now (to avoid that we will find certificate issue, and try to fix it in context of these changes, while it will be unrelated). Likely, we will encounter some other problems, so I preferred to just skip this improvement for now, to reduce issues scope

Reply:
ok